File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.cpp
class Solution {
public:
    int earliestFullBloom(vector<int>& plantTime, vector<int>& growTime) {
        
        int n = plantTime.size();
        vector<pair<int, int>> plant(n);
        for(int i=0; i<n; i++){
            plant[i].first = growTime[i];
            plant[i].second = plantTime[i];
        }
        
        sort(plant.begin(), plant.end(), greater<pair<int, int>>());
        
        int time=0;
        int maxi=0;
        
        for(int i=0; i<n; i++){
            
            time += plant[i].second;
            maxi = max(maxi, time + plant[i].first);
        }
        
        return maxi;
    }
};

File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.java
class Solution {
    private class Plant{
        int plantTime;
        int growTime;
        
        Plant(int plantTime, int growTime){
            this.plantTime = plantTime;
            this.growTime = growTime;
        }
    }
    
    public int earliestFullBloom(int[] plantTime, int[] growTime) {
        
        List<Plant> plant = new ArrayList<>();
        for(int i=0; i<plantTime.length; i++){
            plant.add(new Plant(plantTime[i], growTime[i]));
        }
        
        Collections.sort(plant, (a, b) -> (b.growTime - a.growTime));
        
        int time=0;
        int max=0;
        
        for(int i=0; i<plant.size(); i++){
            time += plant.get(i).plantTime;
            max = Math.max(max, time + plant.get(i).growTime);
        }
        
        return max;
    }
}